docs/operator-manual/config-management-plugins.md

```diff
@@ -503,3 +507,29 @@ spec:
     args: ["sample args"]
   preserveFileMode: true
 ```
+
+##### Provide Git Credentials
+
+By default, the config management plugin is responsible for providing its own credentials to additional Git repositories
+that may need to be accessed during manifest generation. The reposerver has these credentials available in its git creds
+store, to allow the plugin to access these credentials, you can set `provideGitCreds` to `true` in the plugin spec:
+
+!!! warning
+    Make sure you trust the plugin you are using. If you set `provideGitCreds` to `true` then the plugin will receive
+    credentials used to clone the source Git repository.
```

Comment:
I think we probably need a bit more detail about how exactly the credentials are provided to the CMP. That'll 1) help the CMP author debug and 2) help them understand how to secure the creds. The downside is that any details we write into the docs may become out of date if we make major changes to how the creds work. I think that's a worthwhile cost.

Also, just to confirm, this is limited to git creds, right? We're definitely not sending over any Helm creds?

Reply:
Thanks for your input here @crenshaw-dev. I'll add some notes as to how this works, technically it is sending across the `git.Creds` object in the environment of the cmp execution. I'll add that to this document.

Because this uses the `git.Creds` object, it does not include helm credentials.